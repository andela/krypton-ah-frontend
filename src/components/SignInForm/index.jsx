import React from 'react';
import { Form, Checkbox, Button } from 'semantic-ui-react';

class SignIn extends React.Component {
  handleChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }

  handleSubmit(event) {
    event.preventDefault();
  }

  render() {
    const { email, password } = this.state;
    return (
      <div>
        <Form onSubmit={this.handleSubmit.bind(this)}>
          <Form.Field>
            <input
            type="email"
            required
            name="email"
            value={email}
            placeholder="Email Address"
            onChange={this.handleChange.bind(this)}
          />
          </Form.Field>
          <Form.Field>
            <input
            type="password"
            required
            name="password"
            value={password}
            placeholder="Password"
            onChange={this.handleChange.bind(this)}
          />
          </Form.Field>
          <Form.Field>
            <Checkbox label="Remember me" />
          </Form.Field>
          <Button onClick={this.handleSubmit.bind(this)} type="submit" basic fluid huge="true">Sign In</Button>
        </Form>
      </div>
    );
  }
}

export default SignIn;
