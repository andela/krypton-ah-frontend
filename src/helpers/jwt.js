import jwtDecode from 'jwt-decode';
import { authentication } from '../constants';

export const setToken = token => localStorage.setItem(authentication, token);

const getUserIdFromLocalStorage = () => {
  const token = localStorage.getItem(authentication);
  try {
    const { payLoad } = jwtDecode(token);
    return payLoad.id;
  } catch (error) {
    return null;
  }
};

const getToken = localStorage.getItem(authentication);

const config = {
  headers: { Authorization: getToken }
};

export const isUserAuthenticated = () => {
  const token = localStorage.getItem(authentication);
  if (!token) {
    return false;
  }
  try {
    const decodedToken = jwtDecode(token);
    const dateNow = new Date();
    if (decodedToken.exp > (dateNow.getTime() / 1000)) {
      return true;
    }
    return false;
  } catch (error) {
    return false;
  }
};

export { getToken, config, getUserIdFromLocalStorage };
