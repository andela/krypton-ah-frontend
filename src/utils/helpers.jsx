import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';


export function componentSnapshots(componentName) {
  const component = create(componentName);
  expect(component.toJSON()).toMatchSnapshot();
}

export function expectedResult(componenetName, field, value) {
  const wrapper = shallow(componenetName);
  expect(wrapper.find(field)).toHaveLength(value);
}
