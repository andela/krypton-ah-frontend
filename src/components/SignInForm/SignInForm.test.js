import React from 'react';
import SignInForm from '.';
import { componentSnapshots } from '../../utils/helpers';

function checkSignInForm() {
  test('it should matches the snapshot', () => {
    componentSnapshots(<SignInForm />);
  });
}

describe('Modal Footer', () => {
  checkSignInForm();
});
