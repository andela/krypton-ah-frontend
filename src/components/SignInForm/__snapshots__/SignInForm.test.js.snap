// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Footer it should matches the snapshot 1`] = `
<div>
  <form
    className="ui form"
    onSubmit={[Function]}
  >
    <div
      className="field"
    >
      <input
        name="email"
        onChange={[Function]}
        placeholder="Email Address"
        required={true}
        type="email"
        value=""
      />
    </div>
    <div
      className="field"
    >
      <input
        name="password"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
        value=""
      />
    </div>
    <div
      className="field"
    >
      <div
        className="ui checkbox"
        onChange={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <input
          checked={false}
          className="hidden"
          readOnly={true}
          tabIndex={0}
          type="checkbox"
        />
        <label>
          Remember me
        </label>
      </div>
    </div>
    <button
      className="ui basic fluid button"
      huge="true"
      onClick={[Function]}
      type="submit"
    >
      Sign In
    </button>
  </form>
</div>
`;
